src/token.rs

```diff
@@ -301,19 +346,14 @@ mod tests {
     #[test_case("1_234", TokenKind::Integer, "1_234" ; "integer with underscores")]
     #[test_case("1.234", TokenKind::Float, "1.234" ; "simple float")]
     #[test_case(".234", TokenKind::Float, ".234" ; "float with no integer")]
+    #[test_case("1.", TokenKind::Float, "1." ; "integer followed by dot")]
     #[test_case("1.234.5", TokenKind::Float, "1.234" ; "float with extra dot")]
     #[test_case(".234.5", TokenKind::Float, ".234" ; "float with no integer and extra dot")]
-    #[test_case("1e1", TokenKind::Float, "1e1" ; "float exponential")]
-    #[test_case("1e1.", TokenKind::Float, "1e1" ; "float exponential followed by dot")]
-    #[test_case("1e", TokenKind::Integer, "1" ; "float incomplete exponential")]
-    #[test_case("1e-", TokenKind::Integer, "1" ; "float incomplete exponential with sign")]
-    #[test_case("1.3e-10", TokenKind::Float, "1.3e-10" ; "float with all")]
```

Comment:
exponents in floating point numbers are no longer supported by the default tokenizer; they can be handled at the parser/evaluator layer instead, by making `e` a binary operator.